### DISCUSSION TITLE
Production hosting for large number of users

### DISCUSSION THREAD

**Discussion by The-Gk** [miguelgrinberg/Flask-SocketIO#1607] [Q&A]
Sir I read about how to host flask socketio in production from the website. But I have a doubt that if the gunicorn can't increase the number of workers can we host a flask website with flasksocketio for more than 50k requests?
If using nginx can make it happen or any other way?

**Comment by miguelgrinberg** [comment]
Have you seen the documentation on using multiple workers? https://flask-socketio.readthedocs.io/en/latest/#using-multiple-workers

**Comment by The-Gk** [comment]
Sir I readed it but can you show a example of  CLI 
 for  command for hosting a production app? And sir you mentioned in the documentation about using redis or any other in case we use multiple servers. So can you show a example code of using redis for multiple flask socketio servers please?

**Comment by miguelgrinberg** [comment]
I don't have any examples other than those in the documentation, sorry. If you've never done a multiworker deployment then this is going to be hard for you, but if you are familiar with the basics of the deployment, then deploying Socket.IO is similar.

**Comment by The-Gk** [comment]
Sir I have done multiworker deployment but the documentation says that we can use only maximum of one worker with gunicorn when deploying flask socketio apps, sorry if iam wrong sir. If only one worker can be used maximum can we handle more than  10k concurrent users?  Can we make it possible without load balancing sir?

**Comment by miguelgrinberg** [comment]
@Gokul-44 I gave you the link to the section of the documentation that discusses multiple workers. The load balancer in Gunicorn is not flexible enough, so it cannot be used, you have to use a better load balancer, such as nginx.

**Comment by bluthen** [comment]
How many users you can handle seems to depend on a lot of things, like what your server is doing and the resources of the machine it is running on.

If the client is set to use websocket only, gunicorn with multiple workers should work though right? That seems to be what I've found running tests.

**Comment by miguelgrinberg** [comment]
@bluthen Gunicorn with multiple workers still requires a message queue for coordination. Other than that yes, WebSocket connections work, but any client who attempts to connect in the default way will not be able to.

**Comment by The-Gk** [comment]
@miguelgrinberg okay sir. I don't know how to use message queue so can you give a example code sir?

**Comment by miguelgrinberg** [comment]
@Gokul-44 There is documentation for all of this.
